### PR TITLE
indexeddb: add checkBlobSupport()

### LINKED ISSUE #8803: indexeddb adapter does not work on webkit
```
$ BROWSER_CONSOLE=1 COUCH_HOST='http://admin:password@localhost:5984' CLIENT=webkit ADAPTERS=indexeddb npm run test
...
Unhandled error in test page: Error [DataCloneError]: Failed to store record in an IDBObjectStore: BlobURLs are not yet supported.
    at put (http://127.0.0.1:8000/packages/node_modules/pouchdb/dist/pouchdb.indexeddb.js:4280:35)
    at write (http://127.0.0.1:8000/packages/node_modules/pouchdb/dist/pouchdb.indexeddb.js:4280:35)
    at unknown (http://127.0.0.1:8000/packages/node_modules/pouchdb/dist/pouchdb.indexeddb.js:4098:14)
    at forEach ([native code]:0:0)
    at processDocs$$1 (http://127.0.0.1:8000/packages/node_modules/pouchdb/dist/pouchdb.indexeddb.js:4060:17)
    at readDone (http://127.0.0.1:8000/packages/node_modules/pouchdb/dist/pouchdb.indexeddb.js:4041:23)
```

This corresponds to code:

```js
4032   function fetchExistingDocs(txn, docs) {                                       
4033     var fetched = 0;                                                            
4034     var oldDocs = {};                                                           
4035    
4036     function readDone(e) {                                                      
4037       if (e.target.result) {                                                    
4038         oldDocs[e.target.result.id] = e.target.result;                          
4039       }                                                                         
4040       if (++fetched === docs.length) {                                          
4041         processDocs$$1(txn, docs, oldDocs);                                     
4042       }                                                                         
4043     }                                                                           
4044                                                                                 
4045     docs.forEach(function (doc) {                                               
4046       txn.objectStore(DOC_STORE).get(doc.id).onsuccess = readDone;              
4047     });                                                                         
4048   }    
```
which comes from: https://github.com/pouchdb/pouchdb/blob/70bd2afcdc908bc4cb964dde15544ca1eb1f1654/packages/node_modules/pouchdb-adapter-indexeddb/src/bulkDocs.js#L49-L65

ref: safari

### PR DESCRIPTION
**Pull request by alxndrsn**
Add support to the indexeddb adapter for browsers which do not support storing blobs directly in IndexedDB.

In current webkit, this error is seen as:

> Error [DataCloneError]: Failed to store record in an IDBObjectStore: BlobURLs are not yet supported.

Closes #8803

Blocked by:

* https://github.com/pouchdb/pouchdb/pull/8802

### CODE REVIEW

**Comment by alxndrsn** [comment]
I don't think this should be merged until CI tests against webkit.

**Review by garethbowen — COMMENTED** [review]
It looks complicated and I'm not sure I get it. I take it we're only storing this in the meta data because of your comment about not wanting to change function signatures to pass it through?

Does the old idb adapter support webkit, and if so, does it do it in the same way?

File: bin/test-browser.js [hunk 1]
```diff
@@ -161,6 +161,15 @@ async function startTest() {
   const page = await browser.newPage();
 
   page.on('pageerror', err => {
+    if (browserName === 'webkit' && err.toString()
```
[the remark this answers is not in the document]
Comment (alxndrsn) [reply bin/test-browser.js:164]: No - this is the fix from https://github.com/pouchdb/pouchdb/pull/8802/files

**Comment by alxndrsn** [comment]
> Does the old idb adapter support webkit, and if so, does it do it in the same way?

Yes - `checkBlobSupport()` is pulled out of `pouchdb-adapter-idb`.

**Comment by alxndrsn** [comment]
> I take it we're only storing this in the meta data because of your comment about not wanting to change function signatures to pass it through?

Yes.  Also `pouchdb-adapter-idb` does something similar - passing via `api._meta`.  I think it would be less hacky to pass the `metadata` object to any `pouchdb-adapter-indexeddb` function which needs it.

**Review by garethbowen — APPROVED** [review]

**Comment by garethbowen** [comment]
> Yes - checkBlobSupport() is pulled out of pouchdb-adapter-idb.

Got it! Thanks.

**Pull request merged**